File: examples/java/kora-java-scheduling-quartz/src/main/java/io/koraframework/example/scheduling/quartz/CronScheduler.java
package io.koraframework.example.scheduling.quartz;

import io.koraframework.common.annotation.Component;
import io.koraframework.scheduling.quartz.ScheduleWithCron;

@Component
public final class CronScheduler {

    private int state = 0;

    @ScheduleWithCron("* * * ? * * *")
    void schedule() {
        state++;
    }

    public int getState() {
        return state;
    }
}

File: examples/kotlin/kora-kotlin-scheduling-quartz/src/main/kotlin/io/koraframework/kotlin/example/scheduling/quartz/CronScheduler.kt
package io.koraframework.kotlin.example.scheduling.quartz

import io.koraframework.common.annotation.Component
import io.koraframework.common.annotation.Tag
import io.koraframework.scheduling.quartz.ScheduleWithCron
import io.koraframework.scheduling.quartz.ScheduleWithTrigger

@Component
class CronScheduler {
    var state = 0
        private set

    @ScheduleWithCron("* * * ? * * *")
    fun schedule() {
        state++
    }
}

